### ISSUE TITLE
arithmetic_closure test failing on julia 1.6.1

### ISSUE THREAD

**Issue by bcgv** [JuliaArrays/StaticArrays.jl#924]
On julia 1.6.1 this test is failing (3 times):

```
arithmetic_closure: Test Failed at D:\xxx\.julia\dev\StaticArrays\test\arraymath.jl:91
  Expression: #= D:\xxx\.julia\dev\StaticArrays\test\arraymath.jl:91 =# @allocated(arithmetic_closure(T0)) == 0
   Evaluated: 128 == 0
```
It is failing at the occurence 10, 14, 15 which means for types Int64, Float32 and Float64.
However this does not match what I see in the REPL:

```
julia> @allocated(StaticArrays.arithmetic_closure(Float32))
0
```
So maybe something is wrong in the test itself ?

**Comment by mateuszbaran** [comment]
`@allocated` isn't particularly precise so it's very likely a false error.

**Comment by thchr** [comment]
This was fixed by #928 and can be closed now.
